### PR TITLE
for #371: check each subpart for content-transfer-encoding

### PR DESCRIPTION
**Pull request by groovecoder**
Some email senders (e.g., StackOverflow) have specific transfer encodings in each subpart of a multi-part message.

```
Content-Type: text/plain; charset="UTF-8"
Content-Transfer-Encoding: quoted-printable

...

Content-Type: text/html; charset="UTF-8"
Content-Transfer-Encoding: base64
```

So, we need to check each part to see if we should decode it differently than default.

### CODE REVIEW

File: emails/views.py [hunk 1]
```diff
@@ -249,15 +249,11 @@ def _sns_message(message_json):
...
-        # E.g., b64decode(message_payload.get_payload()).decode('utf-8')
-        if message_payload.get_content_type() == 'text/plain':
-            text_content = message_payload.get_payload()
-        if message_payload.get_content_type() == 'text/html':
-            html_content = message_payload.get_payload()
+    email_message = message_from_string(
+        message_json['content'], policy=policy.default
+    )
+
+    text_content, html_content = _get_text_and_html_content(email_message)
```
Comment (groovecoder) [inline emails/views.py:256]: Note: we should probably check `email_message.is_multipart()` before assuming the email has both text and html content.

Comment (groovecoder) [reply emails/views.py:256]: If the email message is NOT multipart, we should probably use `email_message.get_body()` and relay that as whatever content_type it is.

**Comment by groovecoder** [comment]
StackOverflow email registration before these changes

image

After these changes ...

image

**Comment by groovecoder** [comment]
And the plain text version ...

Original:
image

These changes:

image

**Comment by say-yawn** [comment]
Testing on dev. Looks good but the header is not being shown. Created https://github.com/mozilla/fx-private-relay/issues/375 as a follow-up.

File: emails/views.py (hunk 1 above)
Comment (groovecoder) [reply emails/views.py:256]: Filed https://github.com/mozilla/fx-private-relay/issues/377 for this.

**Pull request merged**